code/go/pkg/validator/validator.go

```diff
@@ -0,0 +1,21 @@
+package validator
+
+import (
+	"github.com/elastic/package-spec/code/go/internal/validator"
+)
+
+// ValidateFromPath validates a package located at the given path against the
+// appropriate specification and returns any errors.
+func ValidateFromPath(packageRootPath string) error {
+	pkg, err := validator.NewPackage(packageRootPath)
+	if err != nil {
+		return err
+	}
+
+	spec, err := validator.NewSpec(*pkg.SpecVersion)
```

Comment:
Is there a chance that `pkg.SpecVersion` is nil?

Reply:
It should not happen but I added a guard, just in case, in 569b89b.